src/tlo/methods/cervical_cancer.py

```diff
@@ -0,0 +1,1515 @@
+                    odds_ratio_health_seeking_in_adults=2.00)
+        )
+
+# todo: in order to implement screening for cervical cancer creating a dummy symptom - likely there is a better way
+        self.sim.modules['SymptomManager'].register_symptom(
+            Symptom(name='chosen_via_screening_for_cin_cervical_cancer',
+                    odds_ratio_health_seeking_in_adults=100.00)
+        )
+
+        self.sim.modules['SymptomManager'].register_symptom(
+            Symptom(name='chosen_xpert_screening_for_hpv_cervical_cancer',
+                    odds_ratio_health_seeking_in_adults=100.00)
+        )
+
+
+    def initialise_population(self, population):
+        """Set property values for the initial population."""
+        df = population.props  # a shortcut to the data-frame
+        p = self.parameters
+        rng = self.rng
+
+        # defaults
+        df.loc[df.is_alive, "ce_hpv_cc_status"] = "none"
+        df.loc[df.is_alive, "ce_date_diagnosis"] = pd.NaT
+        df.loc[df.is_alive, "ce_date_treatment"] = pd.NaT
+        df.loc[df.is_alive, "ce_stage_at_which_treatment_given"] = "none"
+        df.loc[df.is_alive, "ce_date_palliative_care"] = pd.NaT
+        df.loc[df.is_alive, "ce_date_death"] = pd.NaT
+        df.loc[df.is_alive, "ce_new_stage_this_month"] = False
+        df.loc[df.is_alive, "ce_stage_at_diagnosis"] = "none"
+        df.loc[df.is_alive, "ce_ever_treated"] = False
+        df.loc[df.is_alive, "ce_cc_ever"] = False
+        df.loc[df.is_alive, "ce_xpert_hpv_ever_pos"] = False
+        df.loc[df.is_alive, "ce_via_cin_ever_detected"] = False
+        df.loc[df.is_alive, "ce_date_cryo"] = pd.NaT
+        df.loc[df.is_alive, 'ce_current_cc_diagnosed'] = False
+        df.loc[df.is_alive, "ce_selected_for_via_this_month"] = False
+        df.loc[df.is_alive, "ce_selected_for_xpert_this_month"] = False
+        df.loc[df.is_alive, "ce_biopsy"] = False
+
+
+        # -------------------- ce_hpv_cc_status -----------
+        # this was not assigned here at outset because baseline value of hv_inf was not accessible - it is assigned
+        # st start of main polling event below
+
+        # -------------------- symptoms, diagnosis, treatment  -----------
+        # For simplicity we assume all these are null at baseline - we don't think this will influence population
+        # status in the present to any significant degree
+
+
+    def initialise_simulation(self, sim):
+        """
+        * Schedule the main polling event
+        * Schedule the main logging event
+        * Define the LinearModels
+        * Define the Diagnostic used
+        * Define the Disability-weights
+        * Schedule the palliative care appointments for those that are on palliative care at initiation
+        """
+
+        # ----- SCHEDULE LOGGING EVENTS -----
+        # Schedule logging event to happen immediately
+        sim.schedule_event(CervicalCancerLoggingEvent(self), sim.date + DateOffset(months=0))
```

Comment:
Why schedule logging event immediately yet polling event is starting a month after? Are we interested in logging defaults also?

Reply:
It seems most of the cancer modules are indeed coded like as you did @andrew-phillips-1 . Not sure what the rationale was. I don't think defaults would be a representation of the population at any point. However, other modules have been coded in such a way that logging starts later. Perhaps logging must start when polling starts or a month after polling judging by this parameter `r_vaginal_bleeding_cc_stage1`

Reply:
@andrew-phillips-1. I think logging should happen after a month. Please start with polling event then logging event i.e. see below
```
        # ----- SCHEDULE MAIN POLLING EVENTS -----
        # Schedule main polling event to happen after a month
        sim.schedule_event(CervicalCancerMainPollingEvent(self), sim.date + DateOffset(months=1))

        # ----- SCHEDULE LOGGING EVENTS -----
        # Schedule logging event to happen after a month
        sim.schedule_event(CervicalCancerLoggingEvent(self), sim.date + DateOffset(months=1))
```

This will also agree to your comment [here](https://github.com/UCL/TLOmodel/blob/416027c84b1e882b10ce0de5dbf2b9339f2d9cb9/src/tlo/methods/cervical_cancer.py#L339-L342)

Reply:
Thanks.  @thewati please remind me about incorporating these things when I am working this after you (or of course you could incorporate them if you wish, which would be very helpful)